### PR TITLE
#229. Implement "Sign Up" screen.

### PR DESCRIPTION
**Pull request by ivan-karavaiev**

### CODE REVIEW

**Review by yevhenii-kanivets — CHANGES_REQUESTED** [review]
Good job overall 👍 It looks accordingly to design, and checks are in place, but when I try to sign up with `test@xorum.io / X0rumTester`, I have the following error:

```
2021-02-09 19:08:30.213657+0200 Codeforces WatchR[671:10810340] [User Defaults] Attempt to set a non-property-list object UserAccount(codeforcesUser=null, email=Test@xorum.io, token=71e70076-7d70-4df7-8d36-edb8da853f47) as an NSUserDefaults/CFPreferences value for key userAccount
2021-02-09 19:08:30.334226+0200 Codeforces WatchR[671:10810340] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'Attempt to insert non-property list object UserAccount(codeforcesUser=null, email=Test@xorum.io, token=71e70076-7d70-4df7-8d36-edb8da853f47) for key userAccount'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff23e3de6e __exceptionPreprocess + 350
	1   libobjc.A.dylib                     0x00007fff512a19b2 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff23ed09d1 _CFThrowFormattedException + 194
	3   CoreFoundation                      0x00007fff23d418d2 _CFPrefsValidateValueForKey + 210
	4   CoreFoundation                      0x00007fff23d41daa -[CFPrefsSource setValues:forKeys:count:copyValues:removeValuesForKeys:count:from:] + 586
	5   CoreFoundation                      0x00007fff23d4208c -[CFPrefsSource setValues:forKeys:count:copyValues:from:] + 28
	6   CoreFoundation                      0x00007fff23d420e3 -[CFPrefsSource setValue:forKey:from:] + 67
	7   CoreFoundation                      0x00007fff23e10ae0 __76-[_CFXPreferences setValue:forKey:appIdentifier:container:configurationURL:]_block_invoke + 64
	8   CoreFoundation                      0x00007fff23e3510b __108-[_CFXPreferences(SearchListAdditions) withSearchListForIdentifier:container:cloudConfigurationURL:perform:]_block_invoke + 443
	9   CoreFoundation                      0x00007fff23e34806 normalizeQuintuplet + 326
	10  CoreFoundation                      0x00007fff23e34f32 -[_CFXPreferences(SearchListAdditions) withSearchListForIdentifier:container:cloudConfigurationURL:perform:] + 114
	11  CoreFoundation                      0x00007fff23e10a79 -[_CFXPreferences setValue:forKey:appIdentifier:container:configurationURL:] + 105
	12  CoreFoundation                      0x00007fff23e14a0c _CFPreferencesSetAppValueWithContainer + 108
	13  Foundation                          0x00007fff25987995 -[NSUserDefaults(NSUserDefaults) setObject:forKey:] + 55
	14  Codeforces WatchR                   0x00000001062ce049 $s17Codeforces_WatchR5PrefsC16writeUserAccount04userF0ySo06CommoneF0C_tF + 169
	15  Codeforces WatchR                   0x00000001062ce0c3 $s17Codeforces_WatchR5PrefsC16writeUserAccount04userF0ySo06CommoneF0C_tFTo + 67
	16  Codeforces WatchR                   0x0000000106581634 objc2kotlin.1201 + 452
	17  Codeforces WatchR                   0x000000010667d8e8 kfun:kotlin.coroutines.native.internal.BaseContinuationImpl#resumeWith(kotlin.Result<kotlin.Any?>){} + 760
	18  Codeforces WatchR                   0x0000000106a217d8 kfun:io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith#internal + 952
	19  Codeforces WatchR                   0x0000000106a2138c kfun:io.ktor.util.pipeline.SuspendFunctionGun.loop#internal + 1756
	20  Codeforces WatchR                   0x0000000106a23e0b kfun:io.ktor.util.pipeline.SuspendFunctionGun.object-1.resumeWith#internal + 379
	21  Codeforces WatchR                   0x000000010667db9c kfun:kotlin.coroutines.native.internal.BaseContinuationImpl#resumeWith(kotlin.Result<kotlin.Any?>){} + 1452
	22  Codeforces WatchR                   0x0000000106a217d8 kfun:io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith#internal + 952
	23  Codeforces WatchR                   0x0000000106a2138c kfun:io.ktor.util.pipeline.SuspendFunctionGun.loop#internal + 1756
	24  Codeforces WatchR                   0x0000000106a23e0b kfun:io.ktor.util.pipeline.SuspendFunctionGun.object-1.resumeWith#internal + 379
	25  Codeforces WatchR                   0x000000010667db9c kfun:kotlin.coroutines.native.internal.BaseContinuationImpl#resumeWith(kotlin.Result<kotlin.Any?>){} + 1452
	26  Codeforces WatchR                   0x0000000106a217d8 kfun:io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith#internal + 952
	27  Codeforces WatchR                   0x0000000106a2138c kfun:io.ktor.util.pipeline.SuspendFunctionGun.loop#internal + 1756
	28  Codeforces WatchR                   0x0000000106a23e0b kfun:io.ktor.util.pipeline.SuspendFunctionGun.object-1.resumeWith#internal + 379
	29  Codeforces WatchR                   0x000000010667db9c kfun:kotlin.coroutines.native.internal.BaseContinuationImpl#resumeWith(kotlin.Result<kotlin.Any?>){} + 1452
	30  Codeforces WatchR                   0x0000000106a217d8 kfun:io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith#internal + 952
	31  Codeforces WatchR                   0x0000000106a2138c kfun:io.ktor.util.pipeline.SuspendFunctionGun.loop#internal + 1756
	32  Codeforces WatchR                   0x0000000106a23e0b kfun:io.ktor.util.pipeline.SuspendFunctionGun.object-1.resumeWith#internal + 379
	33  Codeforces WatchR                   0x000000010667db9c kfun:kotlin.coroutines.native.internal.BaseContinuationImpl#resumeWith(kotlin.Result<kotlin.Any?>){} + 1452
	34  Codeforces WatchR                   0x0000000106a217d8 kfun:io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith#internal + 952
	35  Codeforces WatchR                   0x0000000106a2138c kfun:io.ktor.util.pipeline.SuspendFunctionGun.loop#internal + 1756
	36  Codeforces WatchR                   0x0000000106a23e0b kfun:io.ktor.util.pipeline.SuspendFunctionGun.object-1.resumeWith#internal + 379
	37  Codeforces WatchR                   0x000000010667db9c kfun:kotlin.coroutines.native.internal.BaseContinuationImpl#resumeWith(kotlin.Result<kotlin.Any?>){} + 1452
	38  Codeforces WatchR                   0x00000001068bbe98 kfun:kotlinx.coroutines.DispatchedTask#run(){} + 2680
	39  Codeforces WatchR                   0x00000001068f062a kfun:kotlinx.coroutines.DarwinMainDispatcher.dispatch$lambda-0#internal + 90
	40  Codeforces WatchR                   0x00000001068f09f0 kfun:kotlinx.coroutines.DarwinMainDispatcher.$dispatch$lambda-0$FUNCTION_REFERENCE$44.invoke#internal + 64
	41  Codeforces WatchR                   0x00000001068f0a50 kfun:kotlinx.coroutines.DarwinMainDispatcher.$dispatch$lambda-0$FUNCTION_REFERENCE$44.$<bridge-UNN>invoke(){}#internal + 64
	42  Codeforces WatchR                   0x00000001068f1b69 _6f72672e6a6574627261696e732e6b6f746c696e783a6b6f746c696e782d636f726f7574696e65732d636f7265_knbridge8 + 185
	43  libdispatch.dylib                   0x00007fff520db951 _dispatch_call_block_and_release + 12
	44  libdispatch.dylib                   0x00007fff520dc8cb _dispatch_client_callout + 8
	45  libdispatch.dylib                   0x00007fff520e8cdb _dispatch_main_queue_callback_4CF + 1042
	46  CoreFoundation                      0x00007fff23da1869 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	47  CoreFoundation                      0x00007fff23d9c3b9 __CFRunLoopRun + 2041
	48  CoreFoundation                      0x00007fff23d9b8a4 CFRunLoopRunSpecific + 404
	49  GraphicsServices                    0x00007fff38c05bbe GSEventRunModal + 139
	50  UIKitCore                           0x00007fff49372964 UIApplicationMain + 1605
	51  Codeforces WatchR                   0x00000001062eb89b main + 75
	52  libdyld.dylib                       0x00007fff5211c1fd start + 1
	53  ???                                 0x0000000000000001 0x0 + 1
)
```

File: ios/Codeforces Watcher/Application/features/auth/signUp/AgreementTextView.swift [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        font = Font.textSubheadingBig
+        textColor = Palette.black
+        
+        textAlignment = .left
+        attributedText = agreementText
+    }
+    
+    private func addTermsProperty() {
+        let tag = "terms"
+        let link = "https://docs.google.com/document/d/1hvL5NJ-wEJTEj_lurHkGy0kbtBAwd0w0Q1RIb4fzClg/edit?usp=sharing"
```
[the remark this answers is not in the document]
Comment (ivan-karavaiev) [reply ios/Codeforces Watcher/Application/features/auth/signUp/AgreementTextView.swift:40]: Actually these links are exist in common already.

[the remark this answers is not in the document]
Comment (yevhenii-kanivets) [reply ios/Codeforces Watcher/Application/features/auth/signUp/AgreementTextView.swift:40]: So reuse them 😉

**Review by yevhenii-kanivets — APPROVED** [review]
Thanks for the changes 🙏 

The only issue now is that card doesn't update its state after Sign Up / Sign In, but if you would like to deal with it in another PR, don't hesitate to merge this one right away.

**Pull request merged**